src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderMatchAPIIntegrationTest.kt

```diff
@@ -53,67 +57,171 @@ internal class OffenderMatchControllerAPIIntegrationTest : AbstractTestExecution
         ObjectMapperConfig().jackson2ObjectMapperFactory { _: Type?, _: String? -> objectMapper })
   }
 
-  @Test
-  internal fun `access allowed with ROLE_COMMUNITY`() {
-    given()
-        .auth()
-        .oauth2(jwtAuthenticationHelper.createJwt("ROLE_COMMUNITY"))
-        .contentType(MediaType.APPLICATION_JSON_VALUE)
-        .body("{\"surname\": \"Smith\"}")
-        .post("/match")
-        .then()
-        .statusCode(200)
-  }
+  @Nested
+  inner class `basic operation` {
+    @Test
+    internal fun `access allowed with ROLE_COMMUNITY`() {
+      given()
+          .auth()
+          .oauth2(jwtAuthenticationHelper.createJwt("ROLE_COMMUNITY"))
+          .contentType(MediaType.APPLICATION_JSON_VALUE)
+          .body("{\"surname\": \"Smith\"}")
+          .post("/match")
+          .then()
+          .statusCode(200)
+    }
+
+    @Test
+    internal fun `without ROLE_COMMUNITY access is denied`() {
+      given()
+          .auth()
+          .oauth2(jwtAuthenticationHelper.createJwt("ROLE_BINGO"))
+          .contentType(MediaType.APPLICATION_JSON_VALUE)
+          .body("{\"surname\": \"Smith\"}")
+          .post("/match")
+          .then()
+          .statusCode(403)
+    }
+
+    @Test
+    internal fun `should match when a single offender has all matching attributes`() {
+      loadOffenders(
+          OffenderIdentification(
+              surname = "gramsci",
```

Comment:
A small thing, but there's a lot of code creating test data that doesn't actually relate to the test - might be better to hide comm test data in a function to make it clear which data is driving the test and which is irrelevant?

Reply:
I'm trying to show where you have similar records "this particular one"  will be chosen, so I think in this case the other records are significant. But the price is, as you have spotted,  it is hard to read. I'll have a think about a better way of expressing this since my guess we will end up with 30/40 tests at the end